src/Tribe/Assets.php

```diff
@@ -0,0 +1,338 @@
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			// Now add an action to enqueue the registred assets
+			add_action( $asset->action, array( $this, 'enqueue' ), $asset->priority );
+		}
+	}
+
+	public function enqueue() {
+		foreach ( $this->assets as $asset ) {
+			// Skip if we are not on the correct filter
+			if ( current_filter() !== $asset->action ) {
+				continue;
+			}
+
+			// If here is no action we are just registering this asset
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			/**
+			 * Allows developers to hook-in and prevent an asset from been loaded
+			 *
+			 * @param bool   $enqueue If we should enqueue or not a given asset
+			 * @param object $asset   Which asset we are dealing with
+			 */
+			$enqueue = apply_filters( 'tribe_asset_enqueue', true, $asset );
+
+			/**
+			 * Allows developers to hook-in and prevent an asset from been loaded
+			 *
+			 * Note: When you pass callables on the `$asset->filter` argument this will be hooked here
+			 *
+			 * @param bool   $enqueue If we should enqueue or not a given asset
+			 * @param object $asset   Which asset we are dealing with
+			 */
+			$enqueue = apply_filters( 'tribe_asset_enqueue_' . $asset->slug, $enqueue, $asset );
+
+			if ( ! $enqueue ) {
+				continue;
+			}
+
+			if ( 'js' === $asset->type ) {
+				wp_enqueue_script( $asset->slug );
+
+				// Only localize on JS and if we have data
+				if ( ! empty( $asset->localize ) ) {
+					wp_localize_script( $asset->slug, $asset->localize->name, $asset->localize->data );
+				}
+			} else {
+				wp_enqueue_style( $asset->slug );
+			}
+		}
+	}
+
+	/**
+	 * Returns the path to a minified version of a js or css file, if it exists.
+	 * If the file does not exist, returns false.
+	 *
+	 * @param string $url   The path or URL to the un-minified file.
+	 *
+	 * @return string|false The path/url to minified version or false, if file not found.
+	 */
+	public static function maybe_get_min_file( $url ) {
+		if ( ! defined( 'SCRIPT_DEBUG' ) || SCRIPT_DEBUG === false ) {
+			if ( substr( $url, - 3, 3 ) === '.js' ) {
+				$url = substr_replace( $url, '.min', - 3, 0 );
+			}
+
+			if ( substr( $url, - 4, 4 ) === '.css' ) {
+				$url = substr_replace( $url, '.min', - 4, 0 );
+			}
+		}
+
+		$file = str_replace( WP_CONTENT_URL, WP_CONTENT_DIR, $url );
+		if ( file_exists( $file ) ) {
+			return $url;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Register an Asset and attach a callback to the required action to display it correctly
+	 *
+	 * @param  object   $origin    The main Object for the plugin you are enqueueing the script/style for
+	 * @param  string   $slug      Slug to save the asset
+	 * @param  string   $file      Which file will be loaded, either CSS or JS
+	 * @param  array    $deps      Dependencies
+	 * @param  string   $action    A WordPress Action, needs to happen after: `wp_enqueue_scripts`, `admin_enqueue_scripts`, or `login_enqueue_scripts`
```

Comment:
It took me a few readings to get a sense of what we're doing here: wonder if it might help others to expand on the role of `$action` a little. For instance, cover if it must be set to a valid string (right now the description is suggesting I should supply a hook name, but the signature suggests it's optional).